config/value.go

```diff
@@ -333,10 +333,41 @@ func derefType(t reflect.Type) reflect.Type {
 	return t
 }
 
+func convertPrimitiveValue(value interface{}, targetType reflect.Type) (interface{}, error) {
+	ret, err := convertVal(value, targetType)
+	if ret == nil {
```

Comment:
This is refactored from previous convertValue. The error is bubbled up to decide whether TryAs* method should try additional val.(type) or return false and panic. The most I can do here is to log, but it will be just too verbose at this point since the code panics with "Can't convert to" message.